packages/build/src/plugins_core/types.ts

```diff
@@ -0,0 +1,48 @@
+import { NetlifyConfig } from 'packages/build/types/index.js'
+
+import { NetlifyPluginConstants } from '../core/constants.js'
+import { BufferedLogs } from '../log/logger.js'
+
+type $TSFixme = any
+
+export type CoreStepFunctionArgs = {
+  /**
+   * The absolute process working directory of a build
+   */
+  buildDir: string
+  /**
+   * the directory inside a mono repository where it collects the settings from.
+   * This is the value of the package directory field of the build settings
+   * `undefined` if none is set.
+   */
+  packagePath?: string
+  deployId?: string
+  saveConfig: boolean
+  constants: NetlifyPluginConstants
+  quiet?: boolean
+  debug?: boolean
+  logs?: BufferedLogs
+  systemLog?: (message: unknown) => void
+  edgeFunctionsBootstrapURL?: string
+  // systemLog(...args: any[]): void
+  featureFlags?: Record<string, any>
+
+  netlifyConfig: NetlifyConfig
+  explicitSecretKeys: $TSFixme
+
+  buildbotServerSocket: $TSFixme
+}
+
+export type CoreStepFunction = (args: CoreStepFunctionArgs) => Promise<object>
+export type CoreStepCondition = (args: CoreStepFunctionArgs) => Promise<boolean> | boolean
+
+type Event = `on${'Pre' | '' | 'Post'}${'Build' | 'Dev'}`
```

Comment:
nit: This is super fancy, but I personally would prefer the actual static values listed out, which I find more readable.

Reply:
dang I wanted to be lazy and use cool TypeScript features 😎